torch/utils/_python_dispatch.py

```diff
@@ -85,22 +85,33 @@ def _get_current_dispatch_mode_stack():
     return [_get_dispatch_stack_at(i) for i in range(stack_len)]
 
 def _push_mode(mode, k: Optional[DispatchKey] = None):
-    if k is not None:
-        from torch._ops import push_mode_for_key, get_cached_ops
-        # See Note [Not Caching Per-Dispatch-Key Mode Handlers]
-        # Clear the cache of every op that has been used so far, for this particular key.
-        ks = torch._C._functionality_to_backend_keys(k)
-        for op in get_cached_ops():
-            for key in ks:
-                op._uncache_dispatch(key)
-        push_mode_for_key(k, mode)
-    else:
+    if k is None:
         _push_on_torch_dispatch_stack(mode)
```

Comment:
I think we might just want to go all in and change this API (and the others) to `def _push_mode(Mode, is_pre_dispatch: bool = False)`. The per-dispatch-key-mode stuff is scattered in a lot of places, which is pretty confusing given that it's now dead code :(

Reply:
I modified it a bit to make it little more clear